lib/Platform/Intl/java/com/facebook/hermes/intl/PlatformNumberFormatterICU.java

```diff
@@ -0,0 +1,330 @@
+                symbols.setCurrencySymbol(currencySymbol);
+                decimalFormat.setDecimalFormatSymbols(symbols);
+            }
+        }
+
+        return this;
+    }
+
+    @Override
+    public PlatformNumberFormatterICU setGrouping(boolean mGroupingUsed) {
+        mNumberFormat.setGroupingUsed(mGroupingUsed);
+        return this;
+    }
+
+    @Override
+    public PlatformNumberFormatterICU setMinIntergerDigits(int minimumIntegerDigits) {
+        if (minimumIntegerDigits != -1)
+            mNumberFormat.setMinimumIntegerDigits(minimumIntegerDigits);
+
+        return this;
+    }
+
+    @Override
+    public PlatformNumberFormatterICU setSignificantDigits(IPlatformNumberFormatter.RoundingType roundingType, int minimumSignificantDigits, int maximumSignificantDigits) throws JSRangeErrorException {
+        if (mNumberFormat instanceof android.icu.text.DecimalFormat) {
+            if (roundingType == IPlatformNumberFormatter.RoundingType.SIGNIFICANT_DIGITS) {
+                android.icu.text.DecimalFormat decimalFormat = (android.icu.text.DecimalFormat)mNumberFormat;
+                if (minimumSignificantDigits >= 0)
+                    decimalFormat.setMinimumSignificantDigits(minimumSignificantDigits);
+
+                if (maximumSignificantDigits >= 0) {
+                    if (maximumSignificantDigits < decimalFormat.getMinimumSignificantDigits())
+                        throw new JSRangeErrorException("maximumSignificantDigits should be at least equal to minimumSignificantDigits");
+                    decimalFormat.setMaximumSignificantDigits(maximumSignificantDigits);
+                }
+
+                decimalFormat.setSignificantDigitsUsed(true);
+            }
+        }
+
+        return this;
+    }
+
+    @Override
+    public PlatformNumberFormatterICU setFractionDigits(IPlatformNumberFormatter.RoundingType roundingType, int minimumFractionDigits, int maximumFractionDigits) {
+        if (roundingType == IPlatformNumberFormatter.RoundingType.FRACTION_DIGITS) {
+            if (minimumFractionDigits >= 0)
+                mNumberFormat.setMinimumFractionDigits(minimumFractionDigits);
+
+            if (maximumFractionDigits >= 0)
+                mNumberFormat.setMaximumFractionDigits(maximumFractionDigits);
+
+            if (mNumberFormat instanceof android.icu.text.DecimalFormat) {
+                ((android.icu.text.DecimalFormat)mNumberFormat).setSignificantDigitsUsed(false);
+            }
+        }
+
+        return this;
+    }
+
+    @Override
+    public PlatformNumberFormatterICU setSignDisplay(IPlatformNumberFormatter.SignDisplay signDisplay) {
+        if (mNumberFormat instanceof android.icu.text.DecimalFormat) {
+            android.icu.text.DecimalFormat decimalFormat = (android.icu.text.DecimalFormat)mNumberFormat;
+            android.icu.text.DecimalFormatSymbols symbols = decimalFormat.getDecimalFormatSymbols();
```

Comment:
https://developer.android.com/reference/java/text/DecimalFormat#getDecimalFormatSymbols() says this returns a copy.  Do we have tests that this works correctly?

Reply:
We are modifying the currency symbol in the copy and setting it back to DecimalFormat object calling its "setDecimalFormatSymbols" .. SignDisplay is one attribute which try to do our best with the APIs that we got, but still  we do very poorly .. We've it explicitly mentioned in the tests in HermesIntlNumberFormatTest.java (https://github.com/facebook/hermes/pull/338/files#diff-334df24cf4d5b7537b266fd3c4c4151f)

We've better API support available post from API level 30 for SignDisplay : https://developer.android.com/reference/android/icu/number/NumberFormatter.SignDisplay which we can look into later ..